### PR TITLE
Added ChannelSystem to easier initialize Bootstrap

### PR DESCRIPTION
**Pull request by SplotyCode**
Motivation:
Creating Bootstraps is harder than it must. Many people don't use EPOLL because they dont't know about it or don't want to check it EPOLL is available or if KQUEUE is available. But it also makes it easier to change the Bootstrap to a specific channel type.

Modification:
Added the ChannelSystem enum and Bootstrap#setChannelSystem(ChannelSystem).

Result:
This should make working with netty mush easier for simple projects and big projects. It should also improve performance because you don't need to change LoopGroup and ChannelType based on environment. 

This PR should not break anything and as 100% backward compatibility.
I didn't add KQUEUE but I will add it if you like the basic idea.

The ChannelSystem enum would be allot easier if we could use transport-native.

### CODE REVIEW

**Comment by SplotyCode** [comment]
This would look a lot cleaner if transport-native could be imported. I could make an Interface that has the same methods as the enum and then implement the interface from the enum then we could move the enum to an other Module and use the interface in Bootstrap.

**Comment by Bennett-Lynch** [comment]
I think that you're right to say that the native transport support is too difficult for users to discover.  It should be more obvious and intuitive.  Many people don't use something like EPOLL simply because they don't know that they should.  I, as a user, should not have to write my own `if EPOLL avail use EPOLL, elsif KQUEUE avail use KQUEUE, els NIO`, etc. logic.  If there's not much progress on this PR, I would at least recommend raising an issue so that people can discuss about a better approach.

**Comment by eolivelli** [comment]
This patch is related 

https://github.com/netty/netty/pull/6873

File: transport/src/main/java/io/netty/bootstrap/ChannelSystem.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    private Class<? extends Channel> channelClass;
+    private Class<? extends ServerChannel> serverChannelClass;
+
+    ChannelSystem(Class<? extends Channel> channelClass, Class<? extends ServerChannel> serverChannelClass) {
+        this.channelClass = channelClass;
+        this.serverChannelClass = serverChannelClass;
+    }
+
+    public abstract EventLoopGroup newLoopGroup();
```
Comment (blueswhisper) [inline transport/src/main/java/io/netty/bootstrap/ChannelSystem.java]: No entry for specified thread count of `EventLoopGroup` ?

**Comment by normanmaurer** [comment]
We will need to tackle this somehow for netty 5 but I think for netty 4.x we will just leave it as it is for now... Closing.

**Pull request closed without merging**